examples/pil_compare.py

```diff
@@ -0,0 +1,73 @@
+import timeit
+from argparse import ArgumentParser, Namespace
+from functools import partial
+from pathlib import Path
+
+import numpy as np
+from dicom_preprocessing import load_tiff_f32, load_tiff_u16
+from PIL import Image
+
+
+def read_pil(filepath: Path, fp32: bool = False):
+    """Reads a multi-frame TIFF file using PIL and returns a stacked array."""
+    img = Image.open(filepath)
```

Comment:
This is very minor, but I think this would be slightly cleaner. (Maybe not worth changing...)
```python
def gen_pil_frames(filepath: Path) -> Iterator[Image]:
    # Maybe PIL will have a built-in function to replace this in the future?
    with Image.open(filepath) as img:
        try:
            while True:
                yield img.copy()
                img.seek(img.tell() + 1)  # Move to the next frame
        except EOFError:
            pass  # End of sequence


def get_pil_frames(filepath: Path) -> List[Image]:
    return list(gen_pil_frames(filepath))
    
            
def read_pil(filepath: Path, fp32: bool = False):
    """Reads a multi-frame TIFF file using PIL and returns a stacked array."""
    result = np.stack(get_pil_frames(filepath))
    if fp32:
        delta = np.iinfo(result.dtype).max - np.iinfo(result.dtype).min
        result = result.astype(np.float32) / delta
    return result
```

Reply:
I'd rather leave this as-is since it's just a minor benchmarking script.